### PR TITLE
DT-17898 find forms google analytics implementation 

### PR DESCRIPTION
**Pull request by zacharymorel**
## Description
Added Google Analytics objects. 

## Testing done
Spun up http://localhost:3001/find-forms/?q=caregiver and ran clicks on 
Go To Online Tools,
Form Title Link,
Find Forms Search on component mount,
Find Forms Search on component search.

Ran unit tests and e2e tests. 

## Acceptance criteria
- [x] Event added to the dom datalayer correctly.

## Definition of done
- [x] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by zacharymorel** [comment]
@jonwehausen, here's the PR tag you requested!

File: src/applications/find-forms/containers/SearchForm.jsx [hunk 1]
```diff
@@ -46,7 +67,12 @@ export class SearchForm extends Component {
 
   onSubmitHandler = event => {
     event.preventDefault();
-    this.props.fetchFormsThunk(this.state.query);
+    this.props.fetchFormsThunk(this.state.query).then(forms => {
+      // Derive the total number of pages.
+      const totalPages = Math.ceil(forms.length / MAX_PAGE_LIST_LENGTH);
+
+      return this.recordGAEvent(this.state.query, forms.length, totalPages);
+    });
```
Comment (ncksllvn) [inline src/applications/find-forms/containers/SearchForm.jsx:75]: Same as other comment - if we could do this from the Redux action, then we can save on some repetitious code and wouldn't need the [eslint-override](https://github.com/department-of-veterans-affairs/vets-website/pull/15520/files#diff-4deb3f5adeff7ef1d02f9dc276c2270e5dc8bc0800bb927e3169eaad9d0bfc9dR76)

File: src/applications/find-forms/components/SearchResult.jsx [hunk 2]
```diff
@@ -87,16 +122,26 @@ const SearchResult = ({ form, showFindFormsResultsLinkToFormDetailPages }) => {
...
+            recordGAEvent(`Download VA form ${id} ${pdfLabel}`, url, 'pdf')
+          }
+          {...linkProps}
+        >
           Download VA form {id} {pdfLabel}
         </a>
       </dd>
 
-      {formToolUrl ? (
+      {showFindFormsResultsLinkToFormDetailPages && formToolUrl ? (
```
Comment (ncksllvn) [inline src/applications/find-forms/components/SearchResult.jsx:137]: Is this `showFindFormsResultsLinkToFormDetailPages` directly related to this work? No problem if not, just wondering

File: src/applications/find-forms/components/SearchResult.jsx [hunk 3]
```diff
@@ -37,7 +38,37 @@ export const deriveLatestIssue = (d1, d2) => {
...
+    'search-result-chosen-title': eventTitle, // or 'Download VA form 10-10EZ (PDF)' or 'Go to online tool'
+    'search-result-type': eventType, // populate with 'pdf' if pdf, or 'cta' if "Go to online tool"
+    'search-results-pagination-current-page': currentPage, // populate with the current pagination number at time of result click
+    'search-results-position': currentPositionOnPage, // populate with position on page of result click, beginning with 1 as the first result, number in relation to total results on the page (10 being last with 10 results are shown)
+    'search-results-total-count': totalResultsCount, // populate with the total number of search results at time of click
+    'search-results-total-pages': totalResultsPages, // populate with total number of result pages at time of click
+    'search-selection': 'Find forms', // populate consistently with 'Find forms'
+    'search-results-top-recommendation': undefined, // consistently populate with undefined since there's no top recommendations surfaced here
+    'search-typeahead-enabled': false, // consistently populate with false since there's no type ahead enabled for this search feature
+  });
```
[the remark this answers is not in the document]
Comment (zacharymorel) [reply src/applications/find-forms/components/SearchResult.jsx:65]: Yeah, so many keys!

File: src/applications/find-forms/components/SearchResult.jsx (hunk 2 above)
Comment (zacharymorel) [reply src/applications/find-forms/components/SearchResult.jsx:137]: Yes, I have to add the feature flag in for _some_ existing code which is suppose to be reviewed by VSP collab. I added the code before I was aware of the process yah know?

File: src/applications/find-forms/containers/SearchForm.jsx (hunk 1 above)
Comment (zacharymorel) [reply src/applications/find-forms/containers/SearchForm.jsx:75]: Oh yeah, that's a good abstraction. Can do that.

**Comment by zacharymorel** [comment]
@zacharymorel can you give this another once over?

**Comment by zacharymorel** [comment]
@ncksllvn can you give this another once over?

**Review by ncksllvn — APPROVED** [review]
Lgtm!

**Review by jhonny218 — APPROVED** [review]
LGTM

**Pull request merged**